docs/rfcs/0009-remove-interactors-and-controllers.md

```diff
@@ -0,0 +1,375 @@
+                    },
+                )
+            }
+            currentCategoriesSelections.size == POCKET_CATEGORIES_SELECTED_AT_A_TIME_COUNT -> {
+                state.copy(
+                    pocketStoriesCategoriesSelections = state.pocketStoriesCategoriesSelections
+                        .minusOldest()
+                        .plus(
+                            PocketRecommendedStoriesSelectedCategory(
+                                name = action.categoryName,
+                            )
+                        ),
+                )
+            }
+            else -> {
+                state.copy(
+                    pocketStoriesCategoriesSelections =
+                    state.pocketStoriesCategoriesSelections +
+                            PocketRecommendedStoriesSelectedCategory(action.categoryName)
+                )
+            }
+        }
+        // Selecting a category means the stories to be displayed needs to also be changed.
+        updatedCategoriesState.copy(
+            pocketStories = updatedCategoriesState.getFilteredStories(),
+        )
+    }
+}
+```
+
+---
+
+### Interacting with the storage layer through middlewares
+
+What if a feature requires some async data for its initial state, or needs to write changes to disk? There are no restrictions on running impure methods in middleware, so we can respond to an actions that would changes to the storage layer from middlewares. A good example is [ContainerMiddleware](../../android-components/components/feature/containers/src/main/java/mozilla/components/feature/containers/ContainerMiddleware.kt)
+
+Note that the `InitAction` here subscribes to a `Flow` from the database, which will continue to collect updates from the storage layer and dispatch actions to change the state accordingly.
+
+
+Overall, this should convey the following improvements
+- All state updates are guaranteed to be pure and in a single component, allowing for easy testing and reproduction.
+- All side-effects are logically grouped into middlewares, allowing for testing strategies specific to the type of side-effect.
+- Several indirect abstraction layers are removed, minimizing mental model of code.
+
+---
+
+### Single- vs multi-store patterns and mobile constraints
+
+One big difference between Redux for web and for mobile is that mobile is generally under tighter resource constraints. These constraints are the primary driver for our current usage of `lib-state`, which more closely mirrors the "Flux" pattern instead of the "Redux" pattern. One of the primary differences between the two patterns is simply the number of stores they suggest using. Redux will usually have a single store, while Flux would have multiple. 
+
+Multiple stores allows us to manage memory better. For example, state for Settings will be be automatically collected when a `SettingsStore` falls out of scope, instead of being kept permanently in a global store. 
```

Comment:
```suggestion
Multiple stores allows us to manage memory better. For example, state for a Settings UI screen can be held in a `SettingsStore` which can be cleared from memory (garbage collected) when no longer in that UI context.
```

I think what you're trying to say here is that there is no need to hold screen-specific state in a global store.